spec/models/case/sar/offender_spec.rb

```diff
@@ -211,69 +211,69 @@
 
   describe 'number_exempt_pages' do
     context 'invalid values' do
-      it 'errors when string represents a float' do
-        kase = build(:offender_sar_case, number_exempt_pages: "24.45")
+      it 'errors when float is used' do
+        kase = build(:offender_sar_case, number_exempt_pages: 24.45)
         expect(kase).not_to be_valid
-        expect(kase.errors[:number_exempt_pages]).to eq ['must be a positive whole number']
+        expect(kase.errors[:number_exempt_pages]).to eq(['must be a positive whole number'])
       end
-      
-      it 'errors when string represents a word' do
+
+      it 'errors when a word used instead of number' do
         kase = build(:offender_sar_case, number_exempt_pages: "coffee")
         expect(kase).not_to be_valid
-        expect(kase.errors[:number_exempt_pages]).to eq ['must be a positive whole number']
+        expect(kase.errors[:number_exempt_pages]).to eq(['must be a positive whole number'])
       end
-
-      it 'errors when string represents alphanumeric mix of chars' do
-        kase = build(:offender_sar_case, number_exempt_pages: "coffee123")
+      
+      it 'errors when alphanumeric mix of chars is used' do
+        kase = build_stubbed(:offender_sar_case, number_exempt_pages: "coffee123")
         expect(kase).not_to be_valid
-        expect(kase.errors[:number_exempt_pages]).to eq ['must be a positive whole number']
+        expect(kase.errors[:number_exempt_pages]).to eq(['must be a positive whole number'])
       end
 
       it 'errors when string represents negative whole number' do
-        kase = build(:offender_sar_case, number_exempt_pages: "-562")
+        kase = build_stubbed(:offender_sar_case, number_exempt_pages: -562)
         expect(kase).not_to be_valid
-        expect(kase.errors[:number_exempt_pages]).to eq ['must be a positive whole number']
+        expect(kase.errors[:number_exempt_pages]).to eq(['must be a positive whole number'])
       end
     end
 
     context 'valid values' do
       it 'is valid when string represents a positive whole number' do
-        kase = build(:offender_sar_case, number_exempt_pages: "4835")
+        kase = build_stubbed(:offender_sar_case, number_exempt_pages: 4835)
         expect(kase).to be_valid
       end
     end
   end
 
-  describe 'number_dispatched_pages' do
+  describe 'number_final_pages' do
     context 'invalid values' do
-      it 'errors when string represents a float' do
-        kase = build(:offender_sar_case, number_dispatched_pages: "24.45")
+      it 'errors when float is used' do
+        kase = build(:offender_sar_case, number_final_pages: 24.45)
         expect(kase).not_to be_valid
-        expect(kase.errors[:number_dispatched_pages]).to eq ['must be a positive whole number']
+        expect(kase.errors[:number_final_pages]).to eq(['must be a positive whole number'])
       end
```

Comment:
Those test specs do make me wonder... 
I removed those examples for testing those page fields as string in my PR,  as we don't test non-integer values on integer field, non-boolean value for bool type field,... and so on on model level usually ----> although it is definitely no harm if you want to check,  it may be more useful to test non-integer value from controller/view level. But anyway, this is only for my doubt rather than a issue

Reply:
Yeah I saw that in your PR - was wondering what your thoughts were on this. I think you may be right regarding the controller / view tests, but I may leave it for now and then re-look at it later on.